### ISSUE TITLE
Not possible to get nvidia-docker.list on the fly for debian distros and maybe others

### ISSUE THREAD

**Issue by allan4229** [NVIDIA/nvidia-docker#734]
### 1. Not possible to get nvidia-docker.list on the fly
You are relying on a number saying debian(8|9).
But it won't work to everybody.
And I had to guess what to find, for instance.
##### This is how it is described to get the nvidia-docker.list :
```
distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.list
```
##### Basically what tells the release is :
```
. /etc/os-release;echo $ID$VERSION_ID
```
##### testing it
```
. /etc/os-release;echo $ID$VERSION_ID
debian
```

##### what is really needed:
```
curl -s -L https://nvidia.github.io/nvidia-docker/debian9/amd64/nvidia-docker.list
deb https://nvidia.github.io/libnvidia-container/debian9/$(ARCH) /
deb https://nvidia.github.io/nvidia-container-runtime/debian9/$(ARCH) /
deb https://nvidia.github.io/nvidia-docker/debian9/$(ARCH) /
```
Also, I'm not sure what this ```$(ARCH)``` should mean... it is not used at all by the script lines given.
- Of course it expects a subscript... but it seems like something malformed because neither apt-get nor aptitude recognizes it.

#### What happens
Of course it leads to ```404``` .

### 2. Steps to reproduce the issue
- Use a release of debian that hadn't gone to stable yet. (Not officially numbered yet)
- Use a debian installation with more than a single release in the apt lists.

#### For the arch
- Include the machine arch in the script.
 - ```uname -r | grep -Po '[^-]+$'``` gives amd64.
 - ```uname -m``` but it gives x86_64 and would need to alias the path to amd64 for example.

It would give something like :
```
distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
arch=$(uname -r | grep -Po '[^-]+$' || echo amd64)
curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/$arch/nvidia-docker.list
```

#### For the release
1. redirect ```debian``` path to the latest path available.
 - It may be done to all available distros.
2. Allow seeing the nvidia-docker.list file from the [source](https://nvidia.github.io/nvidia-docker/).
 - Currently it is not possible to read the options by hand, I had to guess the path in the url.

#### For the ```$(ARCH)``` thing...
It is not intended to be used in the home page script of course. I just made this topic to make it easy for users who may read this issue :
```
arch=$(uname -r | grep -Po '[^-]+$' || echo amd64) ;
sed -ri 's/\$\(ARCH\)/'"$arch/" /etc/apt/sources.list.d/nvidia-docker.list
```

### 4. Tips
- ```VERSION_ID``` seems to not work for debian or it is just not exported when using many repos.
- ```/etc/debian_version``` may not contain a number, like ```buster``` .

**Comment by flx42** [comment]
> Use a release of debian that hadn't gone to stable yet. (Not officially numbered yet)

Already mentioned here: #677 
It's not officially supported.

>  Of course it expects a subscript... but it seems like something malformed because neither apt-get nor aptitude recognizes it.

What? Please `man sources.list`

**Comment by allan4229** [comment]
Not all distros use dpkg, right?

**Comment by flx42** [comment]
For yum we use `$basearch`: https://github.com/NVIDIA/nvidia-docker/blob/gh-pages/centos7/nvidia-docker.repo

**Comment by allan4229** [comment]
Hmm, that's the page I was looking for.
It could be linked in Readme for us that need some little different configs.

**Comment by allan4229** [comment]
But I'm glad I had a quick answer.

I was trying to help, you know. Sorry for any inconvenience.

Again : My bad for the ```$(ARCH)``` thing, I made a mistake in the list when I was reading it, and it seemed to be malformed. I used it once or twice to be honest.

My real desire was to just ```docker --device=/dev/dri```... but that's nvidia, thought...

Sorry for any inconvenience, just tried to help anybody that could read it.

Thanks for the tips.
